eng/dir.traversal.targets

```diff
@@ -25,6 +25,22 @@
     <Error Condition="'$(MSBuildLastTaskResult)'=='false'" />
   </Target>
 
+  <Target Name="GetFilesToPackage"
```

Comment:
Actually the problem is more likely that you put it in directory.build.targets.  You're likely overriding the "real" one that we need for src projects.

Reply:
In pkgproj what happens is that ProjectReferences get expanded to *all* PackageConfigurations, that happens here: https://github.com/dotnet/arcade/blob/b0a747adb56ba3f6bdd47a7c72e01e01b51edb87/src/Microsoft.DotNet.Build.Tasks.Configuration/src/build/Microsoft.DotNet.Build.Tasks.Configuration.targets#L86-L99

Then we call GetFilesToPackage on each expanded ProjectReference.  That needs to be the GetFilesToPackage target in the packagelibs.targets.  The GetFilesToPackage in dir.traversal.targets was there to support referencing things like .builds files from packages (as we did in 1.0 era).

Net result, remove the GetFilesToPackage from Directory.Build.targets.  You may not even need it here if we no longer reference .builds/.proj from PkgProjs